tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/util/TlaExUtil.scala

```diff
@@ -23,12 +28,16 @@ object TlaExUtil {
       case OperEx(TlaActionOper.prime, NameEx(name)) =>
         used = used + (name + "'")
 
-      // Do not count the fold operator LET-IN itself
-      case OperEx(ApalacheOper.foldSet | ApalacheOper.foldSeq, LetInEx(_, TlaOperDecl(_, _, localBody)),
-              baseExAndCollectionEx @ _*) =>
+      // ignore the names in the auxiliary let-in definition
+      case OperEx(op, LetInEx(_, TlaOperDecl(_, _, localBody)), baseExAndCollectionEx @ _*) if isFold(op) =>
```

Comment:
I don't think it's necessary to introduce an externally defined `isFold`, only to use it in conditional pattern matching, when the built-in pattern matching via `|` works just fine. Especially since `isFold` isn't used elsewhere.

Reply:
It makes it easier to read. Otherwise, the pattern is just too big for my eyes.